app/models/post_type.rb

```diff
@@ -19,6 +19,10 @@ def reactions
     end
   end
 
+  def templates
+    Post.undeleted.where(post_type_id: PostTemplate.post_type_id).where(template_post_type_id: self).to_a
```

Comment:
Won't work how you think it does - you need _either_:

```ruby
.where(template_post_type: self)
```

or

```ruby
.where(template_post_type_id: self.id)
```

Reply:
Somehow it does work (I've tested this after all) but yeah it's better not to rely on it

Reply:
The first one doesn't work (`Mysql2::Error: Unknown column 'posts.template_post_type' in 'where clause'`) but the second one does.

Reply:
That's because you're missing an association - you need to add the `belongs_to :template_post_type` line to the Post model, then the first variant will work.